app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetController.kt

```diff
@@ -59,12 +62,31 @@ class BloodPressureEntrySheetController @Inject constructor(
     return Observable.merge(systolicChanges, diastolicChanges)
   }
 
+  private fun prefillWhenUpdatingABloodPressure(events: Observable<UiEvent>): Observable<UiChange> {
+    return events
+        .ofType<BloodPressureEntrySheetCreated>()
+        .filter { it.openAs == UPDATE_BP }
+        .flatMapSingle { bloodPressureRepository.findOne(it.uuid) }
+        .map { bloodPressure -> { ui: Ui -> ui.updateBpMeasurements(bloodPressure.systolic, bloodPressure.diastolic)} }
+  }
+
   private fun bpValidationsAndSaves(events: Observable<UiEvent>): Observable<UiChange> {
     val imeDoneClicks = events.ofType<BloodPressureSaveClicked>()
 
-    val patientUuids = events
+    val patientUuidFromUpdateBp = events
         .ofType<BloodPressureEntrySheetCreated>()
+        .filter { it.openAs == UPDATE_BP }
+        .flatMapSingle { bloodPressureRepository.findOne(it.uuid) }
         .map { it.patientUuid }
+        .take(1)
+        .cache()
```

Comment:
This method has now been broken up into multiple methods and is no longer valid.

However what was happening was that the database was being queried for the patient ID in two separate places in the same chain, so the `cache()` was added to prevent an unnecessary database call.